File: spec/php/Debug0Test.php
<?php
namespace Kaitai\Struct\Tests;

class Debug0Test extends TestCase {
    public function testDebug0() {
        $r = Debug0::fromFile(self::SRC_DIR_PATH . '/fixed_struct.bin');

        // --debug implies --no-auto-read
        $r->_read();

        $this->assertSame(80, $r->one());
        $this->assertSame(3, count($r->arrayOfInts()));
        $this->assertSame(65, $r->arrayOfInts()[0]);
        $this->assertSame(67, $r->arrayOfInts()[1]);
        $this->assertSame(75, $r->arrayOfInts()[2]);
        $this->assertSame(45, $r->_unnamed2());

        // FIXME: also test --read-pos once it is implemented
    }
}

File: spec/go/debug_0_test.go
package spec

import (
	"runtime/debug"
	"os"
	"testing"
	"github.com/kaitai-io/kaitai_struct_go_runtime/kaitai"
	. "test_formats"
	"github.com/stretchr/testify/assert"
)

func TestDebug0(t *testing.T) {
	defer func() {
		if r := recover(); r != nil {
			debug.PrintStack()
			t.Fatal("unexpected panic:", r)
		}
	}()
	f, err := os.Open("../../src/fixed_struct.bin")
	if err != nil {
		t.Fatal(err)
	}
	s := kaitai.NewStream(f)
	var r Debug0
	err = r.Read(s, &r, &r)
	if err != nil {
		t.Fatal(err)
	}
	assert.EqualValues(t, 80, r.One)
	assert.EqualValues(t, 3, len(r.ArrayOfInts))
	assert.EqualValues(t, 65, r.ArrayOfInts[0])
	assert.EqualValues(t, 67, r.ArrayOfInts[1])
	assert.EqualValues(t, 75, r.ArrayOfInts[2])
	// // FIXME: `_unnamed*` fields do not start with an uppercase letter and thus are not exported
	// assert.EqualValues(t, 45, r._unnamed2)

	// FIXME: also test --read-pos once it is implemented
}
